scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py

```diff
@@ -0,0 +1,234 @@
+    new_mpn = new_mpn_example.format(n=pincount)
+
+    # handle arguments
+    orientation_str = configuration['orientation_options'][orientation]
+    footprint_name = configuration['fp_name_format_string'].format(man=manufacturer,
+        series=series,
+        mpn=mpn, num_rows=number_of_rows, pins_per_row=pincount, mounting_pad = "",
+        pitch=pitch, orientation=orientation_str)
+
+    kicad_mod = Footprint(footprint_name)
+    descr_format_str = "Molex {:s}, old/engineering part number: {:s} example for new part number: {:s}, {:d} Pins ({:s}), generated with kicad-footprint-generator"
+    kicad_mod.setDescription(descr_format_str.format(series_long, mpn, new_mpn, pincount, datasheet))
+    kicad_mod.setTags(configuration['keyword_fp_string'].format(series=series,
+        orientation=orientation_str, man=manufacturer,
+        entry=configuration['entry_direction'][orientation]))
+
+    # calculate working values
+    end_pos_x = (pincount-1) * pitch
+    centre_x = (end_pos_x - start_pos_x) / 2.0
+    nudge = configuration['silk_fab_offset']
+    silk_w = configuration['silk_line_width']
+    fab_w = configuration['fab_line_width']
+
+
+    body_edge={
+        'left':start_pos_x - pitch/2,
+        'right':end_pos_x + pitch/2,
+        'bottom':5
+        }
+    body_edge['top'] = body_edge['bottom']-10.01
+
+    # create pads
+    # kicad_mod.append(Pad(number=1, type=Pad.TYPE_THT, shape=Pad.SHAPE_RECT,
+    #                     at=[0, 0], size=pad_size,
+    #                     drill=drill, layers=Pad.LAYERS_THT))
+
+    optional_pad_params = {}
+    if configuration['kicad4_compatible']:
+        optional_pad_params['tht_pad1_shape'] = Pad.SHAPE_RECT
+    else:
+        optional_pad_params['tht_pad1_shape'] = Pad.SHAPE_ROUNDRECT
+
+    kicad_mod.append(PadArray(initial=1, start=[start_pos_x, 0],
+        x_spacing=pitch, pincount=pincount,
+        size=pad_size, drill=drill,
+        type=Pad.TYPE_THT, shape=Pad.SHAPE_OVAL, layers=Pad.LAYERS_THT,
+        **optional_pad_params))
+
+    # create fab outline
+    kicad_mod.append(RectLine(start=[body_edge['left'], body_edge['top']],\
+        end=[body_edge['right'], body_edge['bottom']], layer='F.Fab', width=fab_w))
+
+    # create silkscreen
+    kicad_mod.append(RectLine(start=[body_edge['left']-nudge, body_edge['top']-nudge],\
+        end=[body_edge['right']+nudge, body_edge['bottom']+nudge], layer='F.SilkS', width=silk_w))
+
+    # pin 1 markers
+    kicad_mod.append(Line(start=[body_edge['left']-0.4, -2.0],\
+        end=[body_edge['left']-0.4, 2.0], layer='F.SilkS', width=silk_w))
+
+    sl=1
+    poly_pin1_marker = [
+        {'x': body_edge['left'], 'y': -sl/2},
+        {'x': body_edge['left'] + sl/sqrt(2), 'y': 0},
+        {'x': body_edge['left'], 'y': sl/2}
+    ]
+    kicad_mod.append(PolygoneLine(polygone=poly_pin1_marker, layer='F.Fab', width=fab_w))
+
+    yr1=body_edge['bottom']+nudge
+    yr2 = yr1 - 1
+    yr3 = yr2 - 0.53
+    if pincount <= 6:
```

Comment:
The slot in the back plate is quite different in this series compared to the other KK series. It always has the same dimension but there are also parts with 2 slots. The position of where to find the slots is in the main table on page 1 of the datasheet.

Reply:
Fixed in 84b21113c8de2851741bfadcff3f46d2ddd73319